File: sign_test.rb
require 'openssl'
require 'base64'

private_key_file = File.read 'private_key.pem'
private_key = OpenSSL::PKey::RSA.new private_key_file, 'password'
public_key = OpenSSL::PKey::RSA.new File.read 'public_key.pem'

message = 'This is my message'
puts "message:\n#{message}\n\n"

def sign_test(message, private_key, public_key)
  signature = private_key.sign OpenSSL::Digest::SHA1.new, message
  signature = Base64.encode64(signature).gsub(/\n/, "") #Base64 encode without newlines

  puts 'signature:'
  puts "#{signature}\n\n"

  valid = public_key.verify OpenSSL::Digest::SHA1.new, Base64.decode64(signature), message

  if valid
    puts 'signature ok!'
  else
    puts "signature didn't match"
  end
end

puts "sign test:\n"
sign_test(message, private_key, public_key)

File: sign_test.php
<?php
$config['private_key'] = 'file://./private_key.pem';
$config['passphrase'] = 'password';
$config['public_key'] = 'file://./public_key.pem';

$private_key = openssl_get_privatekey($config['private_key'], $config['passphrase']);
$public_key = openssl_get_publickey($config['public_key']);

$message = 'This is my message';

echo "message:\n$message\n\n";

echo "sign test:\n";
signTest($message, $private_key, $public_key);

function signTest($message, $private_key, $public_key) {
	$signature = '';

	$ret = openssl_sign($message, $signature, $private_key);
	if($ret === false) {
		die('error signing');
	}
	$signature = base64_encode($signature);

	echo "signature:\n";
	echo $signature . "\n\n";

	// verify signature
	$ret = openssl_verify($message, base64_decode($signature), $public_key);

	switch($ret) {
		case 1:
			echo "signature ok!\n";
		break;
		case -1:
			die("error verifying signature\n");
		break;
		default:
			echo "signature didn't match\n";
		break;
	}
}
